### PR TITLE
Fix grpc destroy bug

### LINKED ISSUE #11851: 多机单卡，跑resnet模型，一个pass后出core
**paddle版本：v0.14
模型：resnet
运行条件：**
2个psserver，2个trainer（每个单卡）
**启动命令：**
```bash
export PADDLE_TRAINING_ROLE=TRAINER 
export PADDLE_PSERVER_PORT=5002 
export PADDLE_TRAINERS=2
export PADDLE_CURRENT_IP=127.0.0.1
python fluid_benchmark.py --model resnet --async_mode \
--data_set flowers \
--iterations 20 \
--device GPU --gpus 8 --batch_size 32 \
--pass_num 1000 \
--update_method pserver
```
运行结果：
```bash

Total examples: 480, total time: 7.25379, 66.17226 examples/sed

Pass: 0, Loss: 7.170465 , Test Accuracy: 0.008301


pure virtual method called
terminate called without an active exception
*** Aborted at 1530260273 (unix time) try "date -d @1530260273" if you are using GNU date ***
PC: @                0x0 (unknown)
*** SIGABRT (@0x15448) received by PID 87112 (TID 0x7ff260863700) from PID 87112; stack trace: ***
    @     0x7ff26043d390 (unknown)
    @     0x7ff260097428 gsignal
    @     0x7ff26009902a abort
    @     0x7ff258a5f92d __gnu_cxx::__verbose_terminate_handler()
    @     0x7ff258a5d996 __cxxabiv1::__terminate()
    @     0x7ff258a5d9e1 std::terminate()
    @     0x7ff258a5e48f __cxa_pure_virtual
    @     0x7ff205e83aa6 grpc::Channel::~Channel()
    @     0x7ff205e83b59 grpc::Channel::~Channel()
    @     0x7ff2050ff539 std::_Sp_counted_base<>::_M_release()
    @     0x7ff205e46d4e paddle::operators::distributed::GRPCClient::~GRPCClient()
    @     0x7ff205e46e61 paddle::operators::distributed::GRPCClient::~GRPCClient()
    @     0x7ff205e4db7a std::unique_ptr<>::~unique_ptr()
    @     0x7ff26009bff8 (unknown)
    @     0x7ff26009c045 exit
    @           0x51dc1f Py_Exit
    @           0x51b1b7 (unknown)
    @           0x51aadd PyErr_PrintEx
    @           0x4309d5 (unknown)
    @           0x493ae2 Py_Main
    @     0x7ff260082830 __libc_start_main
    @           0x4933e9 _start
    @                0x0 (unknown)
```

### PR DESCRIPTION
**Pull request by velconia**
This close #11851

### CODE REVIEW

**Comment by velconia** [comment]
The implementation of grpc paddle patch is [here](https://github.com/velconia/grpc/tree/v1.10.x-paddle), I use a tricky way to solve this problem and I will submit a issue to grpc group

**Comment by typhoonzero** [comment]
Cool, thanks. Can you please put this patch file under paddle repo so and patch this file in `grpc.cmake` so that we can know the change.

**Comment by velconia** [comment]
It's not easy to do this, the code style of grpc was different from the Paddle's

**Comment by typhoonzero** [comment]
Can try stop the linter to check this file

**Comment by velconia** [comment]
I make use of the git apply function to patch the grpc now, it's more clear

File: patches/grpc/fix_too_early_destory.patch [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
++      g_core_codegen_interface->grpc_completion_queue_destroy(cq_);
++	}
+   }
+ 
+   /// Tri-state return for AsyncNext: SHUTDOWN, GOT_EVENT, TIMEOUT.
+diff --git a/include/grpcpp/impl/codegen/grpc_library.h b/include/grpcpp/impl/codegen/grpc_library.h
+index 17c904d71a..a092b2204d 100644
+--- a/include/grpcpp/impl/codegen/grpc_library.h
++++ b/include/grpcpp/impl/codegen/grpc_library.h
+@@ -19,6 +19,8 @@
```
Comment (jacquesqiao) [inline patches/grpc/fix_too_early_destory.patch:29]: Are these code auto-generated?

Comment (velconia) [reply patches/grpc/fix_too_early_destory.patch:29]: It's generated by `git diff` in grpc repo

Comment (jacquesqiao) [reply patches/grpc/fix_too_early_destory.patch:29]: I mean the file `include/grpcpp/impl/codegen/grpc_library.h`

Comment (velconia) [reply patches/grpc/fix_too_early_destory.patch:29]: No, the file was written by grpc authors and will help generating code

**Review by typhoonzero — APPROVED** [review]
LGTM++ Cool

**Pull request merged**